drivers/hmis/spec/requests/hmis/get_project_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative 'hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  before(:all) do
+    cleanup_test_environment
+  end
+  after(:all) do
+    cleanup_test_environment
+  end
+
+  include_context 'hmis base setup'
+
+  let(:view_access_group) { create :view_access_group }
+
+  let(:query) do
+    <<~GRAPHQL
+      query GetProject($id: ID!) {
+        project(id: $id) {
+          #{scalar_fields(Types::HmisSchema::Project)}
+          organization {
+            id
+          }
+        }
+      }
+    GRAPHQL
+  end
+
+  describe 'project query' do
+    before(:each) do
+      hmis_login(user)
+      assign_viewable(view_access_group, p1.as_warehouse, hmis_user)
+    end
+
+    it 'should resolve invalid enum values as INVALID enum' do
+      p1.update(project_type: 300)
+      p1.update(housing_type: 300)
+      p1.update(tracking_method: 300)
+
+      aggregate_failures 'checking response' do
+        response, result = post_graphql({ id: p1.id }) { query }
+        expect(response.status).to eq 200
+        project = result.dig('data', 'project')
+        expect(project).to include(
+          'id' => p1.id.to_s,
+          'projectType' => 'INVALID',
+          'housingType' => 'INVALID',
+          'trackingMethod' => 'INVALID',
+        )
+      end
+    end
+
+    it 'should resolve invalid NoYesMissing values as nil' do
+      p1.update(residential_affiliation: 50)
+      p1.update(continuum_project: 50)
+
+      aggregate_failures 'checking response' do
+        response, result = post_graphql({ id: p1.id }) { query }
+        expect(response.status).to eq 200
+        project = result.dig('data', 'project')
+        expect(project).to include(
+          'id' => p1.id.to_s,
+          'residentialAffiliation' => nil,
+          'continuumProject' => nil,
+        )
+      end
```

Comment:
Invalid values are not surfaced (yet) for `yes_no_missing_field`s, which resolve fields that use HUD list 1.7 (0=no, 1=yes, 99=DNC) as nullable booleans (0=false, 1=true, 99=null). If the field has some other value, it resolves as null.

In order to support invalid values for 1.7 fields, we'll need to change them in the schema from Booleans to Enums. This would require some changes upstream: change boolean usage in form definitions to choice usages; update checkbox component to support 0/1/99/INVALID under the hood.

Reply:
@eanders unless we are comfortable potentially losing data from 1.7 fields, I think we need to get rid of the concept of yes_no_missing_field and replace them with 1.7 Enum fields 😞

Reply:
I _think_ for now, we're ok with losing anything that isn't 0, 1, or 99 from 1.7.  It could become a problem in the future if other options get added, or if something that was a 1.7 gets turned into a larger set.  I don't see that happening without warning though.